js/models/messages.js

```diff
@@ -518,7 +518,15 @@
                                     catch (e) {
                                         handleError(e);
                                     }
-                                }, handleError);
+                                }, function(error) {
+                                    console.log('handleDataMessage: Message', message.idForLogging(), 'was deleted');
+
+                                    if (confirm) {
```

Comment:
Yep, this is entirely for race-condition situations having to do with disappearing messages deleted right before the `fetch()` call - that's the primary reason the `fetch()` would throw an error.

`confirm` really shouldn't be null, but this method is written extremely defensively because we've been having some trouble with it. The more time this method gets in its current form without errors, the more confident I'll feel pulling back some of the really paranoid checks.